src/analysis/convex/topology.lean

```diff
@@ -186,4 +187,35 @@ by simp only [metric.diam, convex_hull_ediam]
   metric.bounded (convex_hull s) ↔ metric.bounded s :=
 by simp only [metric.bounded_iff_ediam_ne_top, convex_hull_ediam]
 
+lemma convex.is_path_connected {s : set E} (hconv : convex s) (hne : s.nonempty) :
+  is_path_connected s :=
+begin
+  refine is_path_connected_iff.mpr ⟨hne, _⟩,
+  intros x y x_in y_in,
+  let f := λ θ : ℝ, x + θ • (y - x),
+  have hf : continuous f, by continuity,
+  have h₀ : f 0 = x, by simp [f],
+  have h₁ : f 1 = y, by { dsimp [f], rw one_smul, abel },
+  have H := hconv.segment_subset x_in y_in,
+  rw segment_eq_image' at H,
+  exact joined_in.of_line hf.continuous_on h₀ h₁ H
+end
+
+@[priority 100]
+instance normed_space.path_connected : path_connected_space E :=
+begin
+  rw path_connected_space_iff_univ,
+  refine is_path_connected_iff.mpr ⟨⟨0, trivial⟩, _⟩,
```

Comment:
Why not apply `convex_univ.is_path_connected` here?

Reply:
I shouldn't have done that at midnight, I forgot my own API.